src/submap.h

```diff
@@ -418,6 +418,35 @@ class maptile_impl
             return *std::prev( sm->get_items( pos() ).cend() );
         }
 
+        /** Returns the uppermost favorite item if there is a favorite item. Otherwise, returns the item with the greatest volume.
+        *
+        * Assumes there is at least one item.
+        * Used to decide which item to render.
+        * TODO: might be slow enough to justify storing the most visible item for each tile? 
+        */
+        const item &get_most_visible_item() const {
+
+            const item *biggest_item = nullptr;
+            units::volume biggest_volume = 0_ml;
+
+            // uppermost item is at end of the list, so iterate backwards so uppermost item is preferred
+            auto items = get_items();
+            // unsigned int i = get_items().size();
+            for( auto it = items.rbegin(); it != items.rend(); it++ ) {
+                const auto &item = *it;
```

Comment:
Avoid using `auto`

Reply:
It's generally acceptable for more complex types such as iterators, though. So only the one inside the loop needs to be changed, in case that wasn't clear.